sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4508,23 +4508,35 @@ def _execute_file_mounts(self, handle: CloudVmRayResourceHandle,
                 continue
 
             storage = cloud_stores.get_storage_from_path(src)
-            if storage.is_directory(src):
-                sync = storage.make_sync_dir_command(source=src,
-                                                     destination=wrapped_dst)
-                # It is a directory so make sure it exists.
-                mkdir_for_wrapped_dst = f'mkdir -p {wrapped_dst}'
-            else:
-                sync = storage.make_sync_file_command(source=src,
-                                                      destination=wrapped_dst)
-                # It is a file so make sure *its parent dir* exists.
-                mkdir_for_wrapped_dst = (
-                    f'mkdir -p {os.path.dirname(wrapped_dst)}')
+            try:
+                if storage.is_directory(src):
+                    sync_cmd = (storage.make_sync_dir_command(
+                        source=src, destination=wrapped_dst))
+                    # It is a directory so make sure it exists.
+                    mkdir_for_wrapped_dst = f'mkdir -p {wrapped_dst}'
+                else:
+                    sync_cmd = (storage.make_sync_file_command(
+                        source=src, destination=wrapped_dst))
+                    # It is a file so make sure *its parent dir* exists.
+                    mkdir_for_wrapped_dst = (
+                        f'mkdir -p {os.path.dirname(wrapped_dst)}')
+            except Exception as e:  # pylint: disable=broad-except
+                logger.error(
+                    f'Failed to fetch from the bucket {src!r} to '
+                    f'remote instance at {dst!r}.\n'
+                    'Error details: '
+                    f'{common_utils.format_exception(e, use_bracket=True)}.')
+                # If 'cmd' was appended to 'symlink_commands' for this sync, we
+                # remove as it failed to sync.
+                if not dst.startswith('~/') and not dst.startswith('/tmp/'):
+                    symlink_commands.pop()
+                continue
```

Comment:
With this, it seems we skip the failed buckets that do not exist, which misaligns with our previous behavior for other cloud buckets. We should probably not skip all errors here. 
```yaml
file_mounts:
  /dst: s3://some-non-exist-bucket-zhwu
```

Also, it seems for an Azure blob that does not exist, unlike the other clouds, it raises error for the the sync command execution below. Can we get it fail here when we call `list_blobs` 
```
file_mounts:
    /test: https://skyeastusa37461fd.blob.core.windows.net/some-non-exist-bucket-zhwu
```
```
I 07-15 22:12:46 backend_utils.py:1336] Syncing (to 1 node): https://skyeastusa37461fd.blob.core.windows.net/some-non-exist-bucket-zhwu -> /test
E 07-15 22:12:47 subprocess_utils.py:84] INFO: Any empty folders will not be processed, because source and/or destination doesn't have full folder support
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] Job 0c4734a3-9317-ad47-5ecc-0978846fbf69 has started
E 07-15 22:12:47 subprocess_utils.py:84] Log file is located at: /home/azureuser/.azcopy/0c4734a3-9317-ad47-5ecc-0978846fbf69.log
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] Cannot perform sync due to error: cannot list files due to reason -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/vsts/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.15.0/azblob/zc_storage_error.go:42
E 07-15 22:12:47 subprocess_utils.py:84] ===== RESPONSE ERROR (ServiceCode=ContainerNotFound) =====
E 07-15 22:12:47 subprocess_utils.py:84] Description=The specified container does not exist.
E 07-15 22:12:47 subprocess_utils.py:84] RequestId:8316c808-001e-005c-0b04-d71755000000
E 07-15 22:12:47 subprocess_utils.py:84] Time:2024-07-15T22:12:47.2300060Z, Details: 
E 07-15 22:12:47 subprocess_utils.py:84]    Code: ContainerNotFound
E 07-15 22:12:47 subprocess_utils.py:84]    GET https://skyeastusa37461fd.blob.core.windows.net/some-non-exist-bucket-zhwu?comp=list&delimiter=%2F&include=metadata&restype=container&se=2024-07-15t23%3A12%3A46z&sig=-REDACTED-&sp=rcwl&sr=c&sv=2024-05-04&timeout=901
E 07-15 22:12:47 subprocess_utils.py:84]    User-Agent: [AzCopy/10.17.0 Azure-Storage/0.15 (go1.19.2; linux)]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Client-Request-Id: [79bd1b04-f229-46c8-5053-44fb7d1bc8c5]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Version: [2020-10-02]
E 07-15 22:12:47 subprocess_utils.py:84]    --------------------------------------------------------------------------------
E 07-15 22:12:47 subprocess_utils.py:84]    RESPONSE Status: 404 The specified container does not exist.
E 07-15 22:12:47 subprocess_utils.py:84]    Content-Length: [225]
E 07-15 22:12:47 subprocess_utils.py:84]    Content-Type: [application/xml]
E 07-15 22:12:47 subprocess_utils.py:84]    Date: [Mon, 15 Jul 2024 22:12:47 GMT]
E 07-15 22:12:47 subprocess_utils.py:84]    Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Client-Request-Id: [79bd1b04-f229-46c8-5053-44fb7d1bc8c5]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Error-Code: [ContainerNotFound]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Request-Id: [8316c808-001e-005c-0b04-d71755000000]
E 07-15 22:12:47 subprocess_utils.py:84]    X-Ms-Version: [2020-10-02]
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] 
E 07-15 22:12:47 subprocess_utils.py:84] 
```

Reply:
The first issue you mentioned is resolved at [f40604b](https://github.com/skypilot-org/skypilot/pull/3032/commits/f40604b332e2389705ab63fb595eaecaadcaec6a). And the second issue is resolved at [e212ea2](https://github.com/skypilot-org/skypilot/pull/3032/commits/e212ea29540cce5e27948e91bf69230e8b1c078d).